viewers/nerf_viewer/bridge_server/visualizer.py

```diff
@@ -0,0 +1,131 @@
+import sys
+from typing import Optional, Union, Dict
+
+import umsgpack
+from threading import Thread
+import zmq
+
+
+class ViewerWindow:
+    context = zmq.Context()  # pylint: disable=abstract-class-instantiated
+
+    def __init__(
+        self,
+        zmq_port: int = 6000,
+        ip_address: str = "127.0.0.1"
+    ):
+        self.zmq_port = zmq_port
+        self.client = self.context.socket(zmq.REQ)
+        zmq_url = f"tcp://{ip_address}:{self.zmq_port}"
+        self.client.connect(zmq_url)
+        self.assert_connected()
+
+    def send(self, command):
+        """
+        Send a message to bridge server
+        """
+        self.client.send_multipart(
+            [
+                command["type"].encode("utf-8"),
+                umsgpack.packb(command)
+            ]
+        )
+
+        return umsgpack.packb(self.client.recv())
+
+    def send_ping(self):
+        """
+        Ping to the bridge server
+        """
+        type = "ping"
+        data = umsgpack.packb({
+            "type": type
+        })
+        self.client.send_multipart(
+            [
+                type.encode("utf-8"),
+                data
+            ]
+        )
+        res = self.client.recv()
+
+        return umsgpack.unpackb(res)
+
+    def timeout_ping(self, timeout_in_sec: int = 12):
+        res = [Exception(f"Failed to connect to bridge server in {timeout_in_sec} seconds.")]
+
+        def wrapper_func():
+            res[0] = self.send_ping()
+
+        thread = Thread(target=wrapper_func, daemon=True)
+
+        try:
+            thread.start()
+            thread.join(timeout_in_sec)
+        except Exception as e:
```

Comment:
In most cases, we do not use `Exception` because it corresponds to too broad of an error category. If we can identify a specific type of error, such as `RuntimeError` or `ValueError`, we should only catch the known and specific types of errors.